### ISSUE TITLE
add referential integrity checks on external id used by nmdc-schema instances

### ISSUE THREAD

**Issue by turbomam** [microbiomedata/nmdc-schema#410]
in addition to format checking

best to use a cache

**Comment by aclum** [comment]
Still needed in linkml but we can do this w/sparql queries now.

**Comment by turbomam** [comment]
Yes, I am working on a [poorly-named GitHub action](https://github.com/microbiomedata/nmdc-schema/blob/main/.github/workflows/migration.yaml) that illustrates how to set up the necessary environment, without being dependent on a recently-refreshed GraphDB.

@eecavanna is also working on a Docker solution

All we have to do now is add helpfully-named SPARQL queries to [assets/sparql](https://github.com/microbiomedata/nmdc-schema/tree/main/assets/sparql)

**Comment by aclum** [comment]
Do we still need this now that we have refscan?

**Comment by eecavanna** [comment]
I don't know what the author of the ticket title meant by "external id":

> add referential integrity checks on external id used by nmdc-schema instances

Refscan does this:
1. Uses the schema to determine (a) which fields of which documents can contain an `id` of a document, and (b) which collection(s) the document having that `id` can reside in
3. Uses the database to check whether, for all such fields, the document does, indeed, reside in one of those collections

If, by "external id", the author meant "an `id` of a document", then I'd be in favor of closing this ticket as resolved.
